packages/pubsub/src/histogram.js

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module pubsub/histogram
+ */
+
+'use strict';
+
+var MIN_VALUE = 10000;
+var MAX_VALUE = 600000;
+
+/**
+ *
+ */
+function Histogram() {
+  this.data = new Map();
+  this.length = 0;
+}
+
+/**
+ *
+ */
+Histogram.prototype.add = function(value) {
+  value = Math.max(value, MIN_VALUE);
+  value = Math.min(value, MAX_VALUE);
+
+  if (!this.data.has(value)) {
+    this.data.set(value, 0);
+  }
+
+  var count = this.data.get(value);
```

Comment:
@callmehiphop I considered that too, but then all your keys will be strings, which is not really what you want (although probably not the end of the world).

After thinking about it more, I do not think this is a concern. I was originally thinking it was a concern because the histogram could be having values added by multiple concurrent callers, but then I remembered that Node is single-threaded. :-)

No action needed.